packages/react-debug-tools/src/__tests__/ReactHooksInspectionIntegration-test.js

```diff
@@ -986,6 +986,168 @@ describe('ReactHooksInspectionIntegration', () => {
     `);
   });
 
+  it('should update isPending returned from useTransition', async () => {
+    const IndefiniteSuspender = React.lazy(() => new Promise(() => {}));
+    let startTransition;
+    function Foo(props) {
+      const [show, setShow] = React.useState(false);
+      const [isPending, _startTransition] = React.useTransition();
+      React.useMemo(() => 'hello', []);
+      React.useMemo(() => 'not used', []);
+
+      // Otherwise we capture the version from the react-debug-tools dispatcher.
```

Comment:
I think we should never return fake values and always return the memoized values from the hooks. Will write a short summary of all the cases I found where we return fake values and why I believe this is problematic.